core/eth/client.go

```diff
@@ -101,6 +106,32 @@ func (client *CallerSubscriberClient) GetERC20Balance(address common.Address, co
 	return numLinkBigInt, nil
 }
 
+const aggregatorAnswerFunctionID = "7e1b4cb0"
+
+var (
+	dec10            = decimal.NewFromInt(10)
+	functionSelector = HexToFunctionSelector(aggregatorAnswerFunctionID)
+)
+
+// GetAggregatorPrice returns the current price at the given address.
+func (client *CallerSubscriberClient) GetAggregatorPrice(address common.Address, precision int32) (decimal.Decimal, error) {
+	var result string
+	args := CallArgs{
+		To:   address,
+		Data: functionSelector.Bytes(),
+	}
+	err := client.Call(&result, "eth_call", args, "latest")
+	if err != nil {
+		return decimal.NewFromInt(0), errors.Wrap(err, fmt.Sprintf("unable to fetch aggregator price from %s", address.Hex()))
+	}
+	i, err := strconv.ParseInt(result, 0, 64)
+	if err != nil {
+		return decimal.NewFromInt(0), errors.Wrap(err, fmt.Sprintf("unable to fetch aggregator price from %s", address.Hex()))
+	}
+	precisionDivisor := dec10.Pow(decimal.NewFromInt32(precision))
+	return decimal.NewFromInt(i).Div(precisionDivisor), nil
+}
```

Comment:
What type of prices can we get from a contract?  If the returned type is Ether, why handle precision and rounding?  A naive implementation would be to use Uint64 to keep accuracy.

Reply:
The `GetAggregatorPrice` method is tied to a specific CL smart contract, where they have a `precision` variable that handles some nuances. @thodges-gh and @se3000 can speak it the specifics of it.

Reply:
Thanks for the clarification, the Ethereum side of the project is still somewhat cryptic to me.